packages/world/src/ObjectType.sol

```diff
@@ -0,0 +1,925 @@
+    return [ObjectTypes.Fuel];
+  }
+
+  function getMovableTypes() internal pure returns (ObjectType[1] memory) {
+    return [ObjectTypes.Player];
+  }
+
+  function getSmartEntityNonBlockTypes() internal pure returns (ObjectType[1] memory) {
+    return [ObjectTypes.Fragment];
+  }
+
+  // Specialized getters
+  // TODO: these are currently part of the codegen, but we should define them in Solidity and import them here
+  function getObjectTypeSchema(ObjectType self) internal pure returns (Vec3[] memory) {
+    if (self == ObjectTypes.Player) {
+      Vec3[] memory playerRelativePositions = new Vec3[](1);
+      playerRelativePositions[0] = vec3(0, 1, 0);
+      return playerRelativePositions;
+    }
+
+    if (self == ObjectTypes.Bed) {
+      Vec3[] memory bedRelativePositions = new Vec3[](1);
+      bedRelativePositions[0] = vec3(0, 0, 1);
+      return bedRelativePositions;
+    }
+
+    if (self == ObjectTypes.TextSign) {
+      Vec3[] memory textSignRelativePositions = new Vec3[](1);
+      textSignRelativePositions[0] = vec3(0, 1, 0);
+      return textSignRelativePositions;
+    }
+
+    return new Vec3[](0);
+  }
+
+  /// @dev Get relative schema coords, including base coord
+  function getRelativeCoords(ObjectType self, Vec3 baseCoord, Direction direction)
+    internal
+    pure
+    returns (Vec3[] memory)
+  {
+    Vec3[] memory schemaCoords = getObjectTypeSchema(self);
+    Vec3[] memory coords = new Vec3[](schemaCoords.length + 1);
+
+    coords[0] = baseCoord;
+
+    for (uint256 i = 0; i < schemaCoords.length; i++) {
+      require(isDirectionSupported(self, direction), "Direction not supported");
+      coords[i + 1] = baseCoord + schemaCoords[i].rotate(direction);
+    }
+
+    return coords;
+  }
+
+  function isDirectionSupported(ObjectType self, Direction direction) internal pure returns (bool) {
+    if (self == ObjectTypes.Bed) {
+      // Note: before supporting more directions, we need to ensure clients can render it
+      return direction == Direction.NegativeX || direction == Direction.NegativeZ;
+    }
+
+    return true;
+  }
+
+  function getRelativeCoords(ObjectType self, Vec3 baseCoord) internal pure returns (Vec3[] memory) {
+    return getRelativeCoords(self, baseCoord, Direction.PositiveZ);
+  }
+
+  function isActionAllowed(ObjectType self, bytes4 sig) internal pure returns (bool) {
+    if (self == ObjectTypes.Player) {
+      return true;
+    }
+
+    if (self == ObjectTypes.Chest) {
+      return sig == ITransferSystem.transfer.selector || sig == IMachineSystem.fuelMachine.selector;
+    }
+
+    return false;
+  }
+
+  function getMaxInventorySlots(ObjectType self) internal pure returns (uint16) {
+    if (self == ObjectTypes.Player) return 36;
+    if (self == ObjectTypes.Chest) return 27;
+    if (self.isPassThrough()) return type(uint16).max;
+    return 0;
+  }
+
+  function getStackable(ObjectType self) internal pure returns (uint16) {
+    if (self.isUniqueObject()) return 1;
```

Comment:
I think before stackable being 0 made sure that you couldnt add invalid objects to your inventory (eg water/fragment), is this still the case?